barretenberg/cpp/src/barretenberg/eccvm/transcript_builder.hpp

```diff
@@ -104,225 +130,397 @@ class ECCVMTranscriptBuilder {
+     * distinct, the slope \f$ \lambda \f$ of the line passing through them is calculated as:
+     *   \f[
+     *   \lambda = \frac{y_2 - y_1}{x_2 - x_1}
+     *   \f]
+     *   This \f$ \lambda \f$ is used to compute the coordinates of the resulting point \f$ R(x_r, y_r) \f$:
+     *   \f[
+     *   x_r = \lambda^2 - x_1 - x_2
+     *   \f]
+     *   \f[
+     *   y_r = \lambda(x_1 - x_r) - y_1
+     *   \f]
+     *
+     * - **Point Doubling (when x1 = x2)**: If the points are the same (i.e., point doubling), the slope \f$ \lambda \f$
+     * is computed as the tangent line at the point: \f[ \lambda = \frac{3x_1^2 + a}{2y_1} \f] where \f$ a \f$ is the
+     * curve parameter. In our case, \f$ a = 0 \f$.
+     *
+     * @param row
+     * @param entry
+     * @param intermediate_accumulator
+     * @param accumulator
+     * @param add_lambda_numerator
+     * @param add_lambda_denominator
+     */
+    static void compute_lambda_numerator_and_denominator(TranscriptRow& row,
+                                                         const VMOperation& entry,
+                                                         const Element& intermediate_accumulator,
+                                                         const Element& accumulator,
+                                                         FF& add_lambda_numerator,
+                                                         FF& add_lambda_denominator)
+    {
+        const Element vm_point = entry.add ? Element(entry.base_point) : intermediate_accumulator;
+
+        const bool vm_infinity = vm_point.is_point_at_infinity();
+        const bool accumulator_infinity = accumulator.is_point_at_infinity();
+
+        // extract coordinates of the current point in ECCOpQueue
+        const FF vm_x = vm_infinity ? 0 : vm_point.x;
+        const FF vm_y = vm_infinity ? 0 : vm_point.y;
+
+        // extract coordinates of the current accumulator
+        const FF accumulator_x = accumulator_infinity ? 0 : accumulator.x;
+        const FF accumulator_y = accumulator_infinity ? 0 : accumulator.y;
+
+        row.transcript_add_x_equal = (vm_x == accumulator_x) || (vm_infinity && accumulator_infinity);
+        row.transcript_add_y_equal = (vm_y == accumulator_y) || (vm_infinity && accumulator_infinity);
+
+        // compute the numerator and denominator of the slope
+        if ((accumulator_x == vm_x) && (accumulator_y == vm_y) && !vm_infinity && !accumulator_infinity) {
+            // Double case (x1 == x2, y1 == y2)
+            add_lambda_denominator = vm_y + vm_y;
+            add_lambda_numerator = vm_x * vm_x * 3;
+        } else if ((accumulator_x != vm_x) && !vm_infinity && !accumulator_infinity) {
+            // General case (x1 != x2)
+            add_lambda_denominator = accumulator_x - vm_x;
+            add_lambda_numerator = accumulator_y - vm_y;
         }
+    }
+    static void batch_invert(std::vector<FF>& inverse_trace_x,
```

Comment:
I think these calls can stay in the original function

Reply:
done